peachjam/views/documents.py

```diff
@@ -69,16 +68,9 @@ def dispatch(self, request, *args, **kwargs):
             restricted_view = RestrictedDocument403View()
             restricted_view.setup(request, *args, **kwargs)
 
-            if not request.user.is_authenticated:
+            if not request.user.has_perm("peachjam.view_document", obj):
                 return restricted_view.dispatch(request, *args, **kwargs)
 
-            if not request.user.is_superuser or not request.user.is_staff:
-                # check if user belongs to a group that has access to the document
-                doc_groups = get_groups_with_perms(obj)
-                user_groups = request.user.groups.all()
-                if not doc_groups.intersection(user_groups):
-                    return restricted_view.dispatch(request, *args, **kwargs)
-
```

Comment:
changed this to use the object permissions directly, anonymous users don't have the permissions so we don't have to check for authentication